physics/discrete_trajectory_body.hpp

```diff
@@ -255,13 +251,8 @@ void DiscreteTrajectory<Frame>::SetDownsampling(
     Length const tolerance) {
   CHECK(is_root());
   CHECK(!downsampling_.has_value());
-  downsampling_.emplace();
-  downsampling_->start_of_dense_timeline = timeline_.begin();
-  downsampling_->dense_intervals =
-      std::distance(downsampling_->start_of_dense_timeline,
-                    timeline_.cend()) - 1;
-  downsampling_->max_dense_intervals = max_dense_intervals;
-  downsampling_->tolerance = tolerance;
+  downsampling_.emplace(
+      timeline_.begin(), max_dense_intervals, tolerance, timeline_);
```

Comment:
Should I be worried that the parameters are not in the right order?

Reply:
oops. Done.